### ISSUE TITLE
Mining continues after node stopped 

### ISSUE THREAD

**Issue by AmberKiso** [iron-fish/ironfish#941]
### What happened?

3 windows command windows:
Window 1> `yarn start start` - normal startup and sync
2> `yarn start status` - returns

```
Version              0.1.20 @ src
Node                 STARTED
Heap Used            24.55 MiB / 26.39 MiB
Memory               240.88 MiB
P2P Network          CONNECTED - In: 0 B/s, Out: 0 B/s, peers 1
Mining               STOPPED - 0 miners, 0 mined
Mem Pool             0 tx
Syncer               IDLE @ 0 blocks per seconds
Blockchain           SYNCED @ HEAD 00000000000ffd6e92cf64aba975637b5dce7155e70947303831dbae709f72d0 (86686)
Workers              STARTED - 0 -> 0 / 6 - 0 jobs Δ, 0 jobs/s
```

3> `yarn start miners:start` - mining begins

2> ^C to get prompt, then `yarn start status` - returns

```
Version              0.1.20 @ src
Node                 STARTED
Heap Used            29.74 MiB / 33.64 MiB
Memory               401.46 MiB
P2P Network          CONNECTED - In: 34.59 KB/s, Out: 8.35 KB/s, peers 19
Mining               STARTED - 1 miners, 0 mined
Mem Pool             0 tx
Syncer               IDLE @ 0 blocks per seconds | avg time to add block 29.06 ms
Blockchain           SYNCED @ HEAD 0000000000036fb569b1b33f740090a47c969f9d3646eabef04dc6754e563bcf (86704)
Workers              STARTED - 0 -> 0 / 6 - 0 jobs Δ, 5.98 jobs/s
```
2> ^C to get prompt, then `yarn start stop` 
Window 1 reports `Not connected to the Iron Fish network` but mining continues in window 3.

2> ^C to get prompt, then `yarn start status`
Eventually I get an error message:

> $ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run status
> Error: connect ETIMEDOUT \\.\pipe\C:\Users\mjolnir\.ironfish\ironfish.ipc
>     at PipeConnectWrap.afterConnect [as oncomplete] (node:net:1161:16)
> error Command failed with exit code 1.

Mining **still** running.

3> ^C to kill mining and get prompt.
2> ^C to get prompt, then `yarn start status` Succeeds and shows correct state

```
Version              0.1.20 @ src
Node                 STOPPED
Heap Used            0 B / 0 B
Memory               0 B
P2P Network          WAITING - In: 0 B/s, Out: 0 B/s, peers 0
Mining               STOPPED - 0 miners, 0 mined
Mem Pool             0 tx
Syncer               STOPPED @ 0 blocks per seconds
Blockchain           SYNCED @ HEAD 000000000006c3a63eb44ee134cec5bf9352869c517d5c8545016c1a75822683 (86705)
Workers              STOPPED
```

I think there's something wrong with how shutdown() in director.ts is called. (This may be the root cause of  #819 BTW.)

I added a log entry to the code:
```
  shutdown(): void {
    this.logger.info(
    `Director shutdown in progress`,
  )
```
The function is called twice when I stopped the node:

> Not connected to the Iron Fish network
> Director shutdown in progress
> Director shutdown in progress

### Version

0.1.20

**Comment by wd021** [comment]
i believe this is solved? @AmberKiso can you confirm if you're still experiencing this.

**Comment by AmberKiso** [comment]
Unfortunately 0.1.21 still has the same behavior.
After `yarn start stop` and `yarn start status` in the recipe:

> 2:52:43 PM - Found 0 errors. Watching for file changes.
> $ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run status
> Error: connect ETIMEDOUT \\.\pipe\C:\Users\mjolnir\.ironfish\ironfish.ipc
>     at PipeConnectWrap.afterConnect [as oncomplete] (node:net:1161:16)
> error Command failed with exit code 1.

And mining is still running.

Any debug flags you want me to turn on?

**Comment by wd021** [comment]
hmm, good for now. i'll ping you if I need anything - I'll try to dig deeper into this next week.

**Comment by AmberKiso** [comment]
@wd021 I took a swing with #993 to see if that would fix the issue. It unfortunately didn't.
https://github.com/AmberKiso/ironfish/tree/investigate-941

**Comment by wd021** [comment]
got to the bottom of this. @AmberKiso https://github.com/iron-fish/ironfish/pull/1011 should fix this.

**Comment by deekerno** [comment]
This was ultimately fixed by #1081.
